src/numpy-stubs/__init__.pyi

```diff
@@ -7054,9 +7054,9 @@ ldexp: Final[_ufunc_2_1] = ...
 left_shift: Final[_ufunc_2_1] = ...
 logaddexp: Final[_ufunc_2_1] = ...
 logaddexp2: Final[_ufunc_2_1] = ...
-logical_and: Final[_ufunc_2_1] = ...
-logical_or: Final[_ufunc_2_1] = ...
-logical_xor: Final[_ufunc_2_1] = ...
+logical_and: Final[_ufunc_2_1[_Call21Bool]] = ...
+logical_or: Final[_ufunc_2_1[_Call21Bool]] = ...
+logical_xor: Final[_ufunc_2_1[_Call21Bool]] = ...
```

Comment:
Similar story with these ones; they should also return an `object_` array if (one of the) inputs are `object_` array-likes:

```pycon
>>> np.logical_or(np.array([0, 1], "O"), np.array([1, 1], "O"))
array([1, 1], dtype=object)
```

(the `ufunc.types` for the `equal`-family of ufuncs is pretty confusing, but actually always returns `bool`, see https://github.com/numpy/numpy/issues/28544)